app/controllers/ops_controller/settings/automate_schedules.rb

```diff
@@ -19,9 +19,13 @@ def automate_schedules_set_vars
 
   def fetch_target_ids
     if params[:target_class] && params[:target_class] != 'null'
-      targets = Rbac.filtered(params[:target_class]).select(:id, :name)
+      targets = Rbac.filtered(params[:target_class]).select(:id, params[:target_class] == 'MiqGroup' ? :description : :name)
```

Comment:
**related to (2):**
I am suggesting to add constant:

```
ATTRIBUTES_FOR_MODELS = [
{MiqGroup => [:description]},
{Tenant => [:name, : use_config_for_attributes]},
,..
]
```

and add method

```
def attributes_for(model)
if exists ATTRIBUTES_FOR_MODELS[model] then return ATTRIBUTES_FOR_MODELS[model] + [:id]
else
return [:id, name ]
end
```
**Note: This is pseudocode, you need to rewrite it to the ruby.**


and then use method in RBAC call
`Rbac.filtered(params[:target_class]).select(attributes_for(*params[:target_class]))`

**related to Tenant:**
it was not working for tenant because `Rbac.filtered(params[:target_class]).select(:id, :name)` was pulling out just two columns `:id, :name` but column `name` also need column `:use_config_for_attributes` so such column needs to be in this enumeration as well.

Reply:
Or just check if there's `:name` and if not then take `:description`.

I'd assume there's less exceptions rather than more for for this so I would not be adding a hash constant just for the one exception that we know about, although it might seems a more robust approach.

Reply:
@martinpovolny So you assume to write the condition for MiqGroup related exception more general. Right? :)